### PR TITLE
feat: Create custom role

### PR DESCRIPTION
**Pull request by Riddhiagrawal001**
## Type of Change

- [ ] Bugfix
- [X] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Checklist

- [x] I ran `npm run re:build`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: src/screens/UserManagement/UserRoleEntry.res [hunk 1]
```diff
@@ -54,6 +74,11 @@ let make = () => {
     if usersData->Array.length === 0 {
```
Comment (JeevaRamu0104) [inline src/screens/UserManagement/UserRoleEntry.res:74]: there is no point in adding this check always usersData is going to be empty

File: src/screens/UserManagement/UserRoleEntry.res [hunk 2]
```diff
@@ -54,6 +74,11 @@ let make = () => {
...
       getUserData()->ignore
     }
+
+    if rolesAvailableData->Array.length == 0 {
+      getRolesAvailable()->ignore
```
Comment (JeevaRamu0104) [inline src/screens/UserManagement/UserRoleEntry.res:79]: rather making a api call over here create a separate component for Role and make a api call over there

File: src/screens/UserManagement/UserRoleEntry.res [hunk 3]
```diff
@@ -110,38 +135,66 @@ let make = () => {
     {
       title: "Roles",
       renderContent: () =>
-        <DefaultLandingPage
-          title="Coming Soon" height="60vh" overriddingStylesTitle="text-3xl font-semibold"
-        />,
+        <div className="mt-5">
+          <LoadedTable
```
Comment (JeevaRamu0104) [inline src/screens/UserManagement/UserRoleEntry.res:139]: create a separate component for roles

File: src/screens/UserManagement/CreateCustomRole.res [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+        subTitle="An invite will be sent to the email addresses to set up a new account"
+      />
+    </RenderIf>
+    <div
+      className={`h-4/5 bg-white relative overflow-y-scroll flex flex-col gap-10 ${paddingClass} ${marginClass}`}>
+      <PageLoaderWrapper screenState>
+        <Form
+          key="invite-user-management"
+          initialValues={initialValuesForForm->JSON.Encode.object}
+          validate={values => UserManagementUtils.validateFormForRoles(values, groupsAdded)}
```
Comment (JeevaRamu0104) [inline src/screens/UserManagement/CreateCustomRole.res:131]: validation is not working as expected.if we enter the role name first and select the permission button is still in disable state.

File: src/screens/UserManagement/CreateCustomRole.res [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+
+    <div className="flex justify-between items-center">
+      <div className="flex flex-col gap-3 items-start col-span-1">
+        <div className="font-semibold"> {heading->React.string} </div>
+        <div className="text-base text-hyperswitch_black opacity-50 flex-1">
+          {description->React.string}
+        </div>
+      </div>
+      <div onClick={_ => handleRemoveOrAdd()}>
+        <CheckBoxIcon
```
Comment (JeevaRamu0104) [inline src/screens/UserManagement/CreateCustomRole.res:27]: move the checkbox to the left side
image

File: src/screens/UserManagement/UserRoleEntry.res [hunk 6]
```diff
@@ -110,38 +135,66 @@ let make = () => {
     {
       title: "Roles",
       renderContent: () =>
-        <DefaultLandingPage
-          title="Coming Soon" height="60vh" overriddingStylesTitle="text-3xl font-semibold"
-        />,
+        <div className="mt-5">
```
Comment (JeevaRamu0104) [inline src/screens/UserManagement/UserRoleEntry.res:138]: image
change the sub label

File: src/screens/UserManagement/CreateCustomRole.res [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+      setPermissionInfo(_ => permissionInfoValue)
+      setScreenState(_ => PageLoaderWrapper.Success)
+    } catch {
+    | _ => setScreenState(_ => PageLoaderWrapper.Error("Something went wrong!"))
+    }
+  }
+
+  React.useEffect0(() => {
+    if permissionInfo->Array.length === 0 {
+      getPermissionInfo()->ignore
```
Comment (JeevaRamu0104) [inline src/screens/UserManagement/CreateCustomRole.res:110]: add else condition and set the screen state to sucess

File: src/screens/UserManagement/UserRoleEntry.res (hunk 1 above)
Comment (Riddhiagrawal001) [reply src/screens/UserManagement/UserRoleEntry.res:74]: Theres a API call for USERS_LIST in ShowUsersData. So if somebody come from there (via clicking back) by that time usersData would already be present.

File: src/screens/UserManagement/UserRoleEntry.res (hunk 6 above)
Comment (Riddhiagrawal001) [reply src/screens/UserManagement/UserRoleEntry.res:138]: Replaced with a placeholder for the time being. Will replace the text once confirmed with product.

**Review by JeevaRamu0104 — APPROVED** [review]

**Pull request merged**